test_scripts/Defects/6_1/3139_02_Stop_streaming_forcibly_for_1st_app_if_2nd_is_activated_and_starts_streaming.lua

```diff
@@ -0,0 +1,142 @@
+--   - send Navi.OnVideoDataStreaming(false) notification to HMI
+--   - send Navi.StopStream(App_1) request to HMI
+--   - send EndService(VIDEO) request to App_1
+--   - start 'StopStreamingTimeout' timeout
+-- 7. App_1 does not stop streaming within 'StopStreamingTimeout' timeout
+-- SDL does (once timeout expires):
+--   - send OnAppInterfaceUnregistered(PROTOCOL_VIOLATION) notification to App_1
+--   - send BC.OnAppUnregistered(unexpectedDisconnect=false) notification to HMI
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("test_scripts/Defects/6_1/common_3139_3140_3142")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Apps Configuration ]]
+common.app.getParams(1).appHMIType = { "PROJECTION" } -- NAVIGATION
+common.app.getParams(2).appHMIType = { "NAVIGATION" }
+
+--[[ Local Variables ]]
+local stopStreamingTimeout = 3000
+local videoDataStoppedTimeout = 1000
+
+--[[ Local Functions ]]
+local log = common.log
+local ts = common.timestamp
+
+local function activateApp2()
+  local cid = common.hmi.getConnection():SendRequest("SDL.ActivateApp", { appID = common.app.getHMIId(2) })
+  log("SDL->HMI ", "SDL.ActivateApp(2)")
+  ts("ActivateApp")
+  common.hmi.getConnection():ExpectResponse(cid)
+  :Do(function()
+      log("SDL->HMI ", "SUCCESS: SDL.ActivateApp")
+    end)
+
+  common.mobile.getSession(2):ExpectNotification("OnHMIStatus",
+    { hmiLevel = "FULL", videoStreamingState = "STREAMABLE" })
+  :Do(function()
+      log("SDL->App2", "OnHMIStatus(FULL)")
+      ts("OnHMIStatus_Full", "app2")
+    end)
+
+  common.mobile.getSession(1):ExpectNotification("OnHMIStatus",
+    { hmiLevel = "LIMITED", videoStreamingState = "NOT_STREAMABLE" })
+  :Do(function()
+      log("SDL->App1", "OnHMIStatus(NOT_STREAMABLE)")
+      ts("OnHMIStatus_Not_Streamable", "app1")
+    end)
+
+  common.hmi.getConnection():ExpectNotification("Navigation.OnVideoDataStreaming", { available = false })
+  :DoOnce(function()
+      log("SDL->HMI ","Navi.OnVideoDataStreaming(false)")
+      ts("Navi.OnVideoDataStreaming", "hmi")
+    end)
+  :Times(AtLeast(1)) -- number of occurences may be >1 due to issue 3142
+
+  common.hmi.getConnection():ExpectRequest("Navigation.StopStream", { appID = common.app.getHMIId(1)})
+  :Do(function(_, data)
+      log("SDL->HMI ","Navi.StopStream")
+      ts("Navi.StopStream", "hmi")
+      common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+      log("HMI->SDL ","SUCCESS: Navi.StopStream")
+    end)
+
+  common.mobile.getSession(1):ExpectEndService(11)
+  :Do(function()
+      log("SDL->App1", "EndService(VIDEO)")
+      ts("EndService", "app1")
+      common.mobile.getSession(1):SendEndServiceAck(11)
+      log("App1->SDL", "EndServiceAck")
+      -- common.stopStreaming(1)
```

Comment:
Please remove commented code.

Reply:
@GetmanetsIrina Please check updates in a755982. I've decided to leave this commented to explicitly show that streaming is still continue.